assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/metrics_processor.py

```diff
@@ -54,127 +38,103 @@ def create_span_tree_from_dataframe_row(self, row: Row) -> Optional[SpanTreeNode
         """
-        completion_count = self.get_value_from_attributes(attributes, COMPLETION_COUNT_KEYS)
-        prompt_count = self.get_value_from_attributes(attributes, PROMPT_COUNT_KEYS)
-        total_count = self.get_value_from_attributes(attributes, TOTAL_COUNT_KEYS)
-        model = self.get_value_from_attributes(attributes, MODEL_KEYS)
-        if model is not None:
-            # TODO:Convert this calculation to pyspark logic.
-            self.total_prompt_count += prompt_count
-            self.total_completion_count += completion_count
-            self.total_token_count += total_count
-            self.total_requests += 1
-
-            if model not in self.model_data:
-                self.model_data[model] = {
-                    "model_name": model,
-                    "model_type": span_type,
-                    "model_prompt_count": 0,
-                    "model_completion_count": 0,
-                    "total_requests": 0
-                }
-            if self.has_completion_count(span_type):
-                self.model_data[model]["model_completion_count"] += completion_count
-            self.model_data[model]["model_prompt_count"] += prompt_count
-            self.model_data[model]["total_requests"] += 1
-
-        if root_input is not None and root_output is not None:
-
-            self.request_data.append({
-                "prompt_count": prompt_count,
-                "input": root_input,
-                "completion_count": completion_count,
-                "output": root_output
-            })
+        tree = self.create_span_tree_from_dataframe_row(row)
+        print("tree before", tree)
+        if tree is None:
+            return []
+        counts_and_requests = []
+        for span in tree:
+            span_type = span.span_type
+            if span_type in INCLUDE_SPAN_TYPE:
+                attributes = json.loads(span.get_node_attribute(attribute_key="attributes"))
+                # in some cases we have LLM/Embedding span with no parent so need to check
+                # before accessing its parent object
+                parent = tree.get_span_tree_node_by_span_id(span.parent_id)
+                parent_input = None
+                parent_output = None
+                if parent is not None:
+                    parent_input = parent.input
+                    parent_output = parent.output
+                # Extract the counts and model information
+                completion_count = self.get_value_from_attributes(attributes, COMPLETION_COUNT_KEYS)
+                prompt_count = self.get_value_from_attributes(attributes, PROMPT_COUNT_KEYS)
+                total_count = self.get_value_from_attributes(attributes, TOTAL_COUNT_KEYS)
+                model = self.get_value_from_attributes(attributes, MODEL_KEYS)
+
+                # Append a tuple with the counts and model to the list
+                if model is not None:
```

Comment:
Is there any case we don't have model information, and we may still need to add the total token account

Reply:
information for aggregation requires model

Reply:
For model level metrics we need that, while for total I don't think it's required